### PR TITLE
Improve skipped question's behavior

### LINKED ISSUE #3: Unexpected behavior using 'when'?
Hi, I'm working with a bunch of list questions using the `when` function, but I'm facing a weird behavior.
For example:
```javascript
const questions = [{
  name: 'order',
  type: 'list',
  message: 'What would you like to order?',
  // choices may be defined as an array or a function that returns an array
  choices: [
    'Coke',
    'Diet Coke',
    'Cherry Coke',
    {name: 'Sprite', disabled: 'Temporarily unavailable'},
    'Water'
  ],
  when: function () {
    return false
  }
}]

enquirer
   .ask(questions)
   .then(answers => {
     console.log(answers)
   })
   .catch(err => {
     console.log(err)
   })
```
and the answers object will be:
```js
{ order: 'Coke' }
```
this is unexpected at least for me, I was expecting the order's answer to be `undefined`
so, I'm not sure if this is the right behavior and if it is, how can I avoid this case?

Thanks

### PR DESCRIPTION
**Pull request by albizures**
override "when" function to avoid setting a value when there is not a default value and the question was not asked.

resolve #3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -14,6 +14,10 @@ function List(question, answers, ui) {
   Radio.apply(this, arguments);
   this.listInitialized = false;
   this.question.type = 'list';
+
+  this.default = question.default;
```
Comment (doowb) [inline index.js]: This isn't necessary, the `default` value is on `this.options.default`

File: index.js [hunk 2]
```diff
@@ -14,6 +14,10 @@ function List(question, answers, ui) {
...
+  this._when = question.when;
```
Comment (doowb) [inline index.js]: This isn't necessary, you can use `Radio.prototype.when` below.

File: index.js [hunk 3]
```diff
@@ -66,6 +70,18 @@ List.prototype.getAnswer = function() {
 };
 
 /**
+ * overriding "when" function to avoid setting a value when there
+ * is not a default value and the question was not asked
+ */
+List.prototype.when = function() {
+  if (this._when && !this._when.apply(this, arguments)) {
```
Comment (doowb) [inline index.js]: It's possible for `this.when` to return a promise, so the results need to be resolved first:

```js
return Promise.resolve(Radio.prototype.when.apply(this, arguments))
  .then(function(when) {
    ...
    return when;
  });
```

File: index.js [hunk 4]
```diff
@@ -66,6 +70,18 @@ List.prototype.getAnswer = function() {
...
+    this.position = this.default === undefined ? -1 : this.default;
```
Comment (doowb) [inline index.js]: To get the correct position, use `this.choices.getIndex(this.options.default)`

**Review by doowb — CHANGES_REQUESTED** [review]
Thanks @albizures!

I have a few comments below based on some of the features that are in `prompt-base` that aren't obvious by looking at `prompt-list` code.

**Comment by doowb** [comment]
Thanks for the PR @albizures! I requested some changes based on conveniences that are already built into `prompt-list` via `prompt-base` and `prompt-choices`.

Also, thanks for linting.

**Comment by albizures** [comment]
hey @doowb, I resolved all of your comments 😃 

Thanks

**Review by doowb — APPROVED** [review]

**Comment by doowb** [comment]
@albizures thanks for making the changes and for the PR! Sorry for the delay in getting back to this.

**Pull request merged**